### PR TITLE
Remove support for pre-3.6 server releases

### PR DESCRIPTION
**Pull request by jyemin**
JAVA-4295

I divided this up into a commit for each release that we're dropping.  So if we ever need to roll back, it's easy to do it for a single release. 

I suggest enabling Hide Whitespace Only Changes in the UI when you review, as many of the lines changed only whitespace based on the removal of conditional statements.

### CODE REVIEW

**Comment by katcharov** [comment]
@jyemin Just curious, what was the process for removing support? Were you able to find all locations by following version checks from some single location? (For example, I see that you have removed methods like `validateReadConcern`, but I did not immediately see why.)

**Comment by jyemin** [comment]
Yes, just following version checks and then removing anything unused as a result of removing the version check and any code branches that were no longer needed.

**Review by katcharov — APPROVED** [review]
lgtm. Nice to see the ~7k lines removed.

File: driver-core/src/main/com/mongodb/internal/connection/CommandMessage.java [hunk 1]
```diff
@@ -177,43 +169,23 @@ protected EncodingMetadata encodeMessageBodyWithMetadata(final BsonOutput bsonOu
...
-
-    private void addDocumentWithPayload(final BsonOutput bsonOutput, final int messageStartPosition) {
-        BsonBinaryWriter bsonBinaryWriter = new BsonBinaryWriter(bsonOutput, getPayloadArrayFieldNameValidator());
-        BsonWriter bsonWriter = new SplittablePayloadBsonWriter(bsonBinaryWriter, bsonOutput, messageStartPosition, getSettings(), payload);
-        BsonDocument commandToEncode = getCommandToEncode();
-        getCodec(commandToEncode).encode(bsonWriter, commandToEncode, EncoderContext.builder().build());
-    }
-
     private int getOpMsgFlagBits() {
         return getOpMsgResponseExpectedFlagBit();
```
Comment (rozza) [inline driver-core/src/main/com/mongodb/internal/connection/CommandMessage.java:190]: Nit - getOpMsgResponseExpectedFlagBit() is only used here - no need for two methods.

**Review by rozza — APPROVED** [review]
LGTM! Amazing how much code can be removed! :partying_face:

File: driver-core/src/main/com/mongodb/internal/connection/CommandMessage.java (hunk 1 above)
Comment (jyemin) [reply driver-core/src/main/com/mongodb/internal/connection/CommandMessage.java]: Inlined

**Pull request merged**